File: backend/app/schemas/briefing.py
from datetime import date, datetime

from pydantic import BaseModel


class BriefArticle(BaseModel):
    """Article in a briefing section with importance context."""
    id: int
    title: str
    description: str | None = None
    url: str
    source_name: str | None = None
    image_url: str | None = None
    topics: list[str] = []
    gemini_summary: str | None = None
    event_type: str | None = None
    severity: str | None = None
    time_sensitivity: str | None = None
    geo_scope: str | None = None
    personal_impact_flags: list[str] = []
    why_it_matters: str | None = None
    must_know_level: str = "normal"
    importance_score: float = 0.0
    interest_score: float = 0.0
    confirmed_sources: int = 1
    published_at: datetime | None = None

    model_config = {"from_attributes": True}


class StorySource(BaseModel):
    """A source article within a story cluster."""
    id: int
    title: str
    url: str
    source_name: str | None = None
    published_at: datetime | None = None


class BriefingStory(BaseModel):
    """A story/topic cluster in the briefing."""
    headline: str
    narrative: str
    why_it_matters: str | None = None
    event_type: str | None = None
    severity: str | None = None
    must_know_level: str = "normal"
    importance_score: float = 0.0
    interest_score: float = 0.0
    sources: list[StorySource] = []
    section_type: str = "for_you"  # "for_you" or "explore"


class BriefingSection(BaseModel):
    """A section in the daily briefing."""
    title: str
    description: str = ""
    stories: list[BriefingStory] = []


class OverviewDomain(BaseModel):
    """A single domain/theme in the at-a-glance overview."""
    domain: str       # e.g. "International", "Economy", "Technology"
    summary: str      # 1-2 sentence summary


class BriefingResponse(BaseModel):
    date: date
    urgent: BriefingSection = BriefingSection(
        title="Urgent",
        description="Critical events you need to know about right now",
    )
    affects_you: BriefingSection = BriefingSection(
        title="Affects You",
        description="News that may impact your daily life",
    )
    interests: BriefingSection = BriefingSection(
        title="Your Interests",
        description="Personalized picks based on your preferences",
    )
    overview: str = ""
    overview_domains: list[OverviewDomain] = []

File: frontend/src/types/briefing.ts
export interface Article {
  id: number;
  title: string;
  description?: string;
  url: string;
  source_name?: string;
  image_url?: string;
  topics: string[];
  gemini_summary?: string;
  recommendation_score: number;
  published_at?: string;
}

export interface BriefArticle {
  id: number;
  title: string;
  description?: string;
  url: string;
  source_name?: string;
  image_url?: string;
  topics: string[];
  gemini_summary?: string;
  event_type?: string;
  severity?: string;
  time_sensitivity?: string;
  geo_scope?: string;
  personal_impact_flags: string[];
  why_it_matters?: string;
  must_know_level: string;
  importance_score: number;
  interest_score: number;
  confirmed_sources: number;
  published_at?: string;
}

export interface BriefingSection {
  title: string;
  description: string;
  articles: BriefArticle[];
}

export interface BriefingResponse {
  date: string;
  urgent: BriefingSection;
  affects_you: BriefingSection;
  interests: BriefingSection;
  overview: string;
}

export interface IndexData {
  name: string;
  symbol: string;
  price: number;
  change: number;
  change_percent: number;
}

export interface WatchlistItem {
  symbol: string;
  name?: string;
  price?: number;
  change?: number;
  change_percent?: number;
}

export interface CalendarEvent {
  title: string;
  start_time?: string;
  end_time?: string;
  location?: string;
  calendar?: string;
}

export interface ReminderItem {
  title: string;
  due_date?: string;
  notes?: string;
  priority: number;
  list_name?: string;
}

export interface Quote {
  text: string;
  author?: string;
}

export interface Inspiration {
  quote?: Quote;
  fun_fact?: string;
  activity?: string;
}

export interface NewsSummarySection {
  topic: string;
  summary: string;
}

export interface NewsSummary {
  overview: string;
  sections: NewsSummarySection[];
}

export interface Briefing {
  date: string;
  news: Article[];
  stocks: {
    indices: IndexData[];
    watchlist: WatchlistItem[];
  };
  calendar: {
    events: CalendarEvent[];
    reminders: ReminderItem[];
    summary: string;
  };
  inspiration: Inspiration;
}
